src/partition/repart.c

```diff
@@ -7115,12 +7123,17 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
                 }
 
-                case ARG_PRIVATE_KEY_URI: {
-                        r = free_and_strdup_warn(&private_key_uri, optarg);
+                case ARG_PRIVATE_KEY_SOURCE:
+                        if (!STARTSWITH_SET(optarg, "file", "provider:", "engine:"))
```

Comment:
this would allow a string like `--private-key-source=filewaldi` through, and it shouldn't be allowed.